src/Datadog.Trace/Agent/Api.cs

```diff
@@ -96,27 +101,36 @@ public async Task<bool> SendTracesAsync(Span[][] traces)
                 // Error handling block
                 if (!success)
                 {
+                    // Exit if we've hit our retry limit
+                    if (retryCount >= retryLimit)
+                    {
+                        // stop retrying
+                        Log.Error(exception, "An error occurred while sending traces to the agent at {0}", _tracesEndpoint);
+                        return false;
+                    }
+
+                    // Before retry delay
                     bool isSocketException = false;
 
-                    if (exception?.InnerException is SocketException se)
+                    Exception innermostException = exception;
+                    while (innermostException?.InnerException != null)
                     {
-                        isSocketException = true;
-                        Log.Error(se, "Unable to communicate with the trace agent at {0}", _tracesEndpoint);
-                        TracingProcessManager.TryForceTraceAgentRefresh();
+                        innermostException = innermostException.InnerException;
                     }
 
-                    if (retryCount >= retryLimit)
+                    if (innermostException is SocketException se)
```

Comment:
can we nest this in the loop and break if we detect SocketException? Not that it won't be innermost, but it seems flexible that way.

Reply:
Good idea, fixed in c4824856a1c5b6323352f525aec7c762d1047247